marlin-build.sh

```diff
@@ -34,15 +34,15 @@ function makeAndCopy
 	if [ ! -d $2 ]; then
 	    git clone https://github.com/Ultimaker/$2.git
 	fi
-    cd $2
-    git checkout $3
-    git pull #Just to be sure.
+    pushd "${2}"
+    git fetch
+    git reset --hard "origin/${3}"
     cd Marlin
     make HARDWARE_MOTHERBOARD=$4 ARDUINO_INSTALL_DIR=${ARDUINO_PATH} ARDUINO_VERSION=${ARDUINO_VERSION} BUILD_DIR=../../build/$1 DEFINES="$5"
     #make HARDWARE_MOTHERBOARD=$4 BUILD_DIR=../../build/$1 DEFINES=$5
     commitid="$(git rev-parse HEAD)"
     echo $1: $commitid >> ../../cura-binary-data/cura/resources/firmware/commit-ids.txt
```

Comment:
Is there something that can be done about the relative paths in here and at line 41?